### DISCUSSION TITLE
Use autocompletion for user in TokenAdmin

### DISCUSSION THREAD

**Discussion by tchaumeny** [encode/django-rest-framework#8533] [Ideas & Suggestions] [answered]
The user table does typically contain a large number of entries and the default select field in the admin can quickly become unusable.

I suggest to set `autocomplete_fields = ("user",)` [here](https://github.com/encode/django-rest-framework/blob/master/rest_framework/authtoken/admin.py#L23), or at least to make it configurable.

**Comment by lovelydinosaur** [comment, marked as answer]
Will the admin automatically include the necessary search-autocomplete URL?
Do we need to be concerned that adding this would break some existing installations that are subclassing and using a custom formfield_override? https://docs.djangoproject.com/en/4.0/ref/contrib/admin/#django.contrib.admin.ModelAdmin.formfield_overrides

**Reply by tchaumeny** [reply]
To your first question I would answer yes as it is working on my project and I didn't have to configure anything.

I would suggest to make it optional with a default to the previous behavior so as to maintain backward-compatibility.

**Reply by lovelydinosaur** [reply]
👉 #8534
